### ISSUE TITLE
plugins resolution from settings convention plugin [possible regression]

### ISSUE THREAD

**Issue by dsvoronin** [gradle/gradle#19852]
Root project plugins section no longer gets its repository settings from convention declared as

`settings.gradle.kts`:

```
pluginManagement {
    includeBuild("build-logic")
    includeBuild("build-logic-settings")
}

plugins {
    id("convention-plugins")
}
```

`build-logic-settings/convnetion-plugins.gradle.kts`:

```
pluginManagement {
   repositories {
       // our proxy settings stuff
   }
}
```

### Expected Behavior

settings convention plugin repositories is used to resolve root project's plugins

### Current Behavior

Resolution only shows implicitly added https://plugins.gradle.org/

### Context

trying to migrate to gradle 7.4 (from 7.3.2), this is a blocker

### Steps to Reproduce 

Checkout: https://github.com/avito-tech/avito-android/tree/settings-74-bug
try any build, like `./gradlew help` will do

rollback to 7.3.x fixes the issue (you can check develop branch )

**Comment by jjohannes** [comment]
Would be interesting to check if this branch of the Gradle build itself still works (e.g. `compileAll`) with `7.4` as it also makes [use of this feature](https://github.com/gradle/gradle/blob/jjohannes/idiomatic/gradlebuild/split/build-logic/build-logic-base/settings-plugin/src/main/kotlin/gradlebuild.repositories.settings.gradle.kts#L22-L32).

**Comment by jbartok** [comment]
Can confirm that it's a regression, fails with the error: 

```
A problem occurred configuring root project 'avito-android-infra'.
> Could not determine the dependencies of null.
   > Could not resolve all task dependencies for configuration ':classpath'.
      > Could not find com.android.tools.build:gradle:7.1.0.
        Searched in the following locations:
          - https://plugins.gradle.org/m2/com/android/tools/build/gradle/7.1.0/gradle-7.1.0.pom
        If the artifact you are trying to retrieve can be found in the repository but without metadata in 'Maven POM' format, you need to adjust the 'metadataSources { ... }' of the repository declaration.
        Required by:
            project :
```

Interrestingly the error with latest `master` branch is different:

```
Execution failed for task ':build-logic-settings:extensions:compileKotlin'.
> Could not resolve all files for configuration ':build-logic-settings:extensions:kotlinCompilerClasspath'.
   > Could not find net.java.dev.jna:jna:5.6.0.
     Required by:
         project :build-logic-settings:extensions > org.jetbrains.kotlin:kotlin-compiler-embeddable:1.6.10
```

**Comment by jbartok** [comment]
> Would be interesting to check if [this branch of the Gradle build](https://github.com/gradle/gradle/pull/15526) itself still works (e.g. `compileAll`) with `7.4` as it also makes [use of this feature](https://github.com/gradle/gradle/blob/jjohannes/idiomatic/gradlebuild/split/build-logic/build-logic-base/settings-plugin/src/main/kotlin/gradlebuild.repositories.settings.gradle.kts#L22-L32).

```
* Where:
Settings file '/Users/jozsef/Work/gradle/settings.gradle.kts' line: 5

* What went wrong:
Error resolving plugin [id: 'gradlebuild.settings-plugins']
> A problem occurred configuring project ':build-logic-base:settings-plugin'.
   > Dependency verification failed for configuration ':build-logic-base:settings-plugin:classpath'
     4 artifacts failed verification:
       - error_prone_annotations-2.3.4.jar (com.google.errorprone:error_prone_annotations:2.3.4) from repository Gradle Central Plugin Repository
       - kotlinx-coroutines-core-jvm-1.5.0.jar (org.jetbrains.kotlinx:kotlinx-coroutines-core-jvm:1.5.0) from repository Gradle Central Plugin Repository
       - semver4j-0.16.4-nodeps.jar (com.github.gundy:semver4j:0.16.4) from repository Gradle Central Plugin Repository
       - gradle-kotlin-dsl-plugins-2.1.7.jar (org.gradle.kotlin:gradle-kotlin-dsl-plugins:2.1.7) from repository Gradle Central Plugin Repository
     If the artifacts are trustworthy, you will need to update the gradle/verification-metadata.xml file by following the instructions at https://docs.gradle.org/7.3.3/userguide/dependency_verification.html#sec:troubleshooting-verification

     Open this report for more details: file:///Users/jozsef/Work/gradle/build/reports/dependency-verification/at-1644404952617/dependency-verification-report.html
```

**Comment by ljacomet** [comment]
@jbartok For the `gradle/gradle` branch, you might want to try with verification disabled

**Comment by fxlae** [comment]
I'm facing the same behaviour after upgrading from 7.3.x. It seems the last working version for me is **7.3.1**.

**Comment by adammurdoch** [comment]
Fixed via https://github.com/gradle/gradle/pull/19961

**Comment by autonomousapps** [comment]
This doesn't work in the following scenario:
```groovy
pluginManagement {
  includeBuild("build-logic-settings") // provides "convention-plugins"
}

plugins {
  id 'convention-plugins'
  id 'foo' version '1.0'
}
```
where plugin `'foo'` should be provided by a repository defined in `'convention-plugins'`. I get this error:

```
* Where:
Settings file '/Users/trobalik/Development/my-project/settings.gradle' line: 129

* What went wrong:
Plugin [id: 'foo', version: '1.0'] was not found in any of the following sources:

- Gradle Core Plugins (plugin is not in 'org.gradle' namespace)
- Included Builds (None of the included builds contain this plugin)
- Plugin Repositories (could not resolve plugin artifact 'foo:foo.gradle.plugin:1.0')
  Searched in the following repositories:
    Gradle Central Plugin Repository
```
The following, however, _does_ work (but is not what I was hoping to achieve)
```
```groovy
pluginManagement {
  includeBuild("build-logic-settings") // provides "convention-plugins"
  repositories {
    maven { ... }
  }
}

plugins {
  id 'convention-plugins'
  id 'foo' version '1.0'
}
```
Adding a repo fixes it, but my whole goal with this pattern was to define the repos just once, in `'convention-plugins'`, and be done with it.

**Comment by jbartok** [comment]
@autonomousapps, I've checked again and the original issue/reproducer is fixed/works fine with both Gradle `7.4.1` & `7.4.2`. Could you pls. open a separate issue & provide a simple reproducer for the problem you are still seeing? Thanks in advance
